src/scaraplate/strategies.py

```diff
@@ -48,6 +48,25 @@
 Pattern = Any  # re.Pattern since py3.7
 
 
+def detect_newline(
+    *file_contents_args: Optional[BinaryIO], default: bytes = b"\n"
+) -> bytes:
```

Comment:
Can we use `NEWLINE= Literal[b"\n", b"\r\n", b"\r"]` to define newline type? This should prevent accidentally use some custom binary suffix in place of newlines and would be a bit better that any arbitrary bytes data.

Reply:
1. `typing.Literal` appeared in Python 3.8, so it is absent in 3.6 and 3.7 which we still support.

2. The current implementation doesn't explicitly state which byte sequences are treated as newlines. This is up to `readline` to deside (which in turn adheres to [PEP 278](https://www.python.org/dev/peps/pep-0278/)). So I would prefer to avoid this enumeration.

3. We could've added a `typing.NewType` specifically for newlines, but I'd argue that it would add any value. Keyword-only arguments in this case IMHO provide enough protection against accidental errors.

Reply:
1. typing_extension wouldn't help there? If it's not - resolve the discussion.
2. Good point, but my idea what about that the actually newline byte sequences are well known and defined, so there is no problem of such.
3. NewType is certainly not a helper here. Literal looks better one since it may be shared and strictly limited.

Reply:
1. It probably would, but it is a new external dependency.
2. I would agree, but I don't see a reason to duplicate this knowledge here.